module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,62 @@
+#include "compression.hpp"
+#include <iostream>
+
+void printMap(const Bitmap& bitmap)
+{
+    for (const auto line : bitmap)
+    {
+        for (const auto pixel : line)
+        {
+            std::cout << pixel;
+        }
+        std::cout << "\n";
+    }
+}
+
+CompressedBitmap compressLine(const std::array<uint8_t, width>& line)
+{
+    CompressedBitmap result;
+    result.reserve(width*height);
+
+    for (const auto pixel : line)
+    {
+        if(!result.empty() and result.back().first == pixel)
+        {
+            result.back().second++;
+        }
+        else
+        {
+            result.push_back(std::make_pair(pixel,1));
+        }
+    }
+
+    result.shrink_to_fit();
+    return result;
+}
+
+CompressedBitmap compressGrayscale(const Bitmap& bitmap)
+{
+    CompressedBitmap result;
+    result.reserve(height*width);
+    CompressedBitmap tmp;
+    tmp.reserve(width);
```

Comment:
Actually you don't need reserve here memory for tmp. Because when you assign result vector to tmp, it will reserve as many memories as needed.

`temp = vec(/*which sie is 50*/) // temp will reserve memory exactly for 50 elements. And you don't need to call shirnk_to_fit(). `